### ISSUE TITLE
Stuck at training

### ISSUE THREAD

**Issue by sarahESL** [Kaushalya/medclip#2]
When I run `run_medclip.sh`, everything initiates perfectly but when it gets to run the training, it gets stuck and does not pass the `0%` progress!

```
loading weights file https://huggingface.co/openai/clip-vit-base-patch32/resolve/main/pytorch_model.bin from cache at /home/**me**/.cache/huggingface/transformers/8a82711445c5200c2b4fd30739df371f5b3ce2d7e316418d58dd290bae1c1cc8.dabcc684421296ebcdafd583a4415c1757ae007787f2d0e17b87482d9b8cf197
Loading PyTorch weights from /home/**me**/.cache/huggingface/transformers/8a82711445c5200c2b4fd30739df371f5b3ce2d7e316418d58dd290bae1c1cc8.dabcc684421296ebcdafd583a4415c1757ae007787f2d0e17b87482d9b8cf197
PyTorch checkpoint contains 151,277,440 parameters.
Some weights of the model checkpoint at openai/clip-vit-base-patch32 were not used when initializing FlaxCLIPModel: {('vision_model', 'embeddings', 'position_ids'), ('text_model', 'embeddings', 'position_ids')}
- This IS expected if you are initializing FlaxCLIPModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing FlaxCLIPModel from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
All the weights of FlaxCLIPModel were initialized from the model checkpoint at openai/clip-vit-base-patch32.
If your task is similar to the task the model of the checkpoint was trained on, you can already use FlaxCLIPModel for predictions without further training.
text_config_dict is None. Initializing the CLIPTextConfig with default values.
vision_config_dict is None. initializing the CLIPVisionConfig with default values.
08/04/2021 09:57:44 - INFO - absl -   A polynomial schedule was set with a non-positive `transition_steps` value; this results in a constant schedule with value `init_value`.
/home/**me**/.local/lib/python3.9/site-packages/jax/lib/xla_bridge.py:364: UserWarning: jax.host_count has been renamed to jax.process_count. This alias will eventually be removed; please update your code.
  warnings.warn(
/home/**me**/.local/lib/python3.9/site-packages/jax/lib/xla_bridge.py:351: UserWarning: jax.host_id has been renamed to jax.process_index. This alias will eventually be removed; please update your code.
  warnings.warn(
08/04/2021 09:57:45 - INFO - __main__ -   ***** Running training *****
08/04/2021 09:57:45 - INFO - __main__ -     Num examples = 65420
08/04/2021 09:57:45 - INFO - __main__ -     Num Epochs = 40
08/04/2021 09:57:45 - INFO - __main__ -     Instantaneous batch size per device = 64
08/04/2021 09:57:45 - INFO - __main__ -     Total train batch size (w. parallel & distributed) = 64
08/04/2021 09:57:45 - INFO - __main__ -     Total optimization steps = 40880
Epoch ... (1/40):   0%|                                                                                                                                                | 0/40 [00:00<?, ?it/s2021-08-04 10:00:17.638335: E external/org_tensorflow/tensorflow/compiler/xla/service/slow_operation_alarm.cc:55]                                                     | 0/1022 [00:00<?, ?it/s]
********************************
Very slow compile?  If you want to file a bug, run with envvar XLA_FLAGS=--xla_dump_to=/tmp/foo and attach the results.
Compiling module pmap_train_step.103881
********************************
                                                                              
```

**Comment by Kaushalya** [comment]
Could you mention details of your environment?
1. Device used for training
2. Jax and Flax versions
3. Transformer version/ if it was installed from Transformers master branch

Did you create your own dataset.json files using [this notebook](https://github.com/Kaushalya/medclip/blob/master/src/medclip/utils/roco_dataset.ipynb)? I have forgotten to mention this step in the README file.

**Comment by sarahESL** [comment]
Thanks for the prompt response.
1. Geforce GTX 1080 ti
2. jax version: 0.2.18 , flax version: 0.3.4
3. Transformer version 4.9.1

Yes I created my own dataset.json s, although I wrote a script myself. But the json format is compliant with those in in your `data` folder and the image paths are correct. I can share the script here, if necessary.

**Comment by Kaushalya** [comment]
Thanks for the details. I haven't trained this model on a GPU yet. I could test it only on CPU and TPUs.
I'll check if I can recreate the issue on a GPU.

**Comment by sarahESL** [comment]
How should I run it only on CPU? and how much resource does it need? like how much RAM?

**Comment by Kaushalya** [comment]
Sorry, I meant I have tested the training script on the CPU on a sample of the dataset. The current model is trained on 8 TPUs.
I don't think it's feasible to train the complete dataset on a single CPU. 
Meanwhile, could you try a smaller batch size?

**Comment by sarahESL** [comment]
I see.
Yeah I did try with smaller batch size (16) and smaller number of workers for `DataLoader` (16), but no hope! When it reaches the `Training...:   0%| `, the GPU memory usage stays at `239MiB` and it usage about `7GiB` of CPU RAM, but nothing further happens and it gets stuck here.
Thanks for your responses anyhow! :slightly_smiling_face:

**Comment by sarahESL** [comment]
Can you maybe report the converged loss value (after fine-tuning with ROCO)?

**Comment by Kaushalya** [comment]
Sorry, I was on vacation. You can find the training curves here https://huggingface.co/flax-community/medclip-roco/tensorboard. I will check the training issue this weekend.

**Comment by sarahESL** [comment]
Oh right! It's also on the repo's readme! Somehow I missed it! :see_no_evil:

**Comment by Kaushalya** [comment]
On my side, I figured out that the issue is with CUDA toolkit. Apparently, JAX is not working with CUDA 11.4. Downgrading it to 11.2 and upgrading JAX to the latest version fixed the issue. Can you make sure that you're not using CUDA 11.4?

![image](https://user-images.githubusercontent.com/101580/130321474-888b5ecb-78c0-4a60-909f-ac2c4a5ee7cd.png)

**Comment by sarahESL** [comment]
Mine is 11.3. I will checkout whether downgrading helps, thanks.
